packages/grid/src/vaadin-grid-selection-column-mixin.js

```diff
@@ -160,7 +151,7 @@ export const GridSelectionColumnMixin = (superClass) =>
           if (!this._grid.selectedItems.length) {
             this.selectAll = false;
             this._indeterminate = false;
-          } else if (this.__arrayContains(this._grid.selectedItems, items)) {
+          } else if (this._grid._areAllSelected(items)) {
```

Comment:
Oh, I just realized that we probably don't need `__arrayContains` at all.

```suggestion
          } else if (items.every((item) => this._grid._isSelected(item)) {
```

Reply:
Sure, let's go with that.